homeassistant/components/climate/ecobee.py

```diff
@@ -218,10 +209,8 @@ def turn_away_mode_off(self):
 
     def set_temperature(self, **kwargs):
         """Set new target temperature."""
-        if kwargs.get(ATTR_TEMPERATURE) is not None:
-            temperature = kwargs.get(ATTR_TEMPERATURE)
-            low_temp = int(temperature)
-            high_temp = int(temperature)
+        if kwargs.get(ATTR_TEMPERATURE) is None:
```

Comment:
This doesn't make any sense. It will now return if temperature is None so the next if statement will never happen.